src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -57,6 +57,18 @@ function getUnmaskedContext(workInProgress : Fiber) : Object {
 }
 exports.getUnmaskedContext = getUnmaskedContext;
 
+function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
+  if (isContextConsumer(workInProgress)) {
```

Comment:
Right, but we don’t need the `instance` to tell if it’s a context consumer or not.
Or do we?

I thought that maybe we can make `getMaskedContext` return `null` for non-context consumers. Then we’d know for sure that we don’t need to call `cacheContext`.

Not saying it’s the best way, just a proposal.